notte/browser/node_type.py

```diff
@@ -405,10 +407,13 @@ class NotteNode:
     id: str | None
     role: NodeRole | str
     text: str
-    subtree_ids: list[str] = field(init=False, default_factory=list)
     children: list["NotteNode"] = field(default_factory=list)
     attributes_pre: NodeAttributesPre = field(default_factory=NodeAttributesPre)
     attributes_post: NotteAttributesPost | None = None
+    subtree_ids: list[str] = field(init=False, default_factory=list)
+    _subtree_chars: int = -1  # length of formatted text for this node+children
```

Comment:
Why not use `None` for default value ?

Reply:
Because it is annoying to have to deal with based-pyright type checks when something can be of two different types. Here at least it's always an integer, and my condition is that that thing is positive, that's it.